flow/raster_cache.cc

```diff
@@ -39,8 +46,24 @@ void RasterCacheResult::draw(DlCanvas& canvas, const DlPaint* paint) const {
              std::abs(bounds.height() - image_->dimensions().height()) <= 1);
   canvas.TransformReset();
   flow_.Step();
-  canvas.DrawImage(image_, {bounds.fLeft, bounds.fTop},
-                   DlImageSampling::kNearestNeighbor, paint);
+  if (is_root_canvas || !rtree_) {
+    canvas.DrawImage(image_, {bounds.fLeft, bounds.fTop},
+                     DlImageSampling::kNearestNeighbor, paint);
+  } else {
+    // On some platforms RTree from overlay layers is used for unobstructed
+    // platform views and hit testing. To preserve the RTree raster cache must
+    // paint individual rects instead of the whole image.
+    auto rects = rtree_->searchAndConsolidateRects(kGiantRect);
+
+    canvas.Translate(bounds.fLeft, bounds.fTop);
+
+    SkRect rtree_bounds = rtree_->bounds();
+    for (auto rect : rects) {
+      rect.offset(-rtree_bounds.fLeft, -rtree_bounds.fTop);
```

Comment:
This could also be factored into the translate above.

Reply:
This affects the `src` position (where we copy from the image), so I don't quite see how.